src/index.ts

```diff
@@ -100,26 +97,27 @@ const isPromise = <T>(value: any): value is Promise<T> =>
   (typeof value === "object" || typeof value === "function") &&
   typeof value.then === "function";
 
-export const combineValidators = <T, ErrorMessage extends string = string>(
-  ...fns: (ValidatorFn<T, ErrorMessage> | false)[]
-): ValidatorFn<T, ErrorMessage> => (value) => {
+export const combineValidators = <T, ErrorMessage = string>(
+  ...fns: (ValidateFn<T, ErrorMessage> | false)[]
+): ValidateFn<T, ErrorMessage> => (value) => {
   const [fn, ...nextFns] = fns;
 
   if (fn) {
-    const error = fn(value);
-    if (isPromise(error)) {
-      return error.then((err) => {
-        if (err) {
-          return err;
+    const result = fn(value);
+
+    if (isPromise(result)) {
+      return result.then((error) => {
+        if (error) {
```

Comment:
maybe we change to `error != null` like you did below with `result`

Reply:
Ah indeed, good catch!